integrationtest/spring/src/test/java/org/hibernate/search/test/integration/spring/injection/SpringInjectionIT.java

```diff
@@ -0,0 +1,121 @@
+		assertThat( search.apply( "au revoir" ) ).onProperty( "id" ).containsOnly( entity2.getId() );
+
+		dao.delete( entity );
+		assertThat( search.apply( "bonjour" ) ).onProperty( "id" ).isEmpty();
+		assertThat( search.apply( "hello" ) ).onProperty( "id" ).isEmpty();
+		assertThat( search.apply( "hallo" ) ).onProperty( "id" ).isEmpty();
+		assertThat( search.apply( "au revoir" ) ).onProperty( "id" ).containsOnly( entity2.getId() );
+	}
+
+	@Test
+	public void injectedClassBridge() {
+		Function<String, List<EntityWithSpringAwareBridges>> search = dao::searchClassBridge;
+
+		assertThat( search.apply( "bonjour" ) ).onProperty( "id" ).isEmpty();
+		assertThat( search.apply( "hello" ) ).onProperty( "id" ).isEmpty();
+		assertThat( search.apply( "hallo" ) ).onProperty( "id" ).isEmpty();
+		assertThat( search.apply( "au revoir" ) ).onProperty( "id" ).isEmpty();
+
+		EntityWithSpringAwareBridges entity = new EntityWithSpringAwareBridges();
+		entity.setInternationalizedValue( InternationalizedValue.HELLO );
+		dao.create( entity );
+		assertThat( search.apply( "bonjour" ) ).onProperty( "id" ).containsOnly( entity.getId() );
+		assertThat( search.apply( "hello" ) ).onProperty( "id" ).containsOnly( entity.getId() );
+		assertThat( search.apply( "hallo" ) ).onProperty( "id" ).containsOnly( entity.getId() );
+		assertThat( search.apply( "au revoir" ) ).onProperty( "id" ).isEmpty();
+
+		EntityWithSpringAwareBridges entity2 = new EntityWithSpringAwareBridges();
+		entity2.setInternationalizedValue( InternationalizedValue.GOODBYE );
+		dao.create( entity2 );
+		assertThat( search.apply( "bonjour" ) ).onProperty( "id" ).containsOnly( entity.getId() );
+		assertThat( search.apply( "hello" ) ).onProperty( "id" ).containsOnly( entity.getId() );
+		assertThat( search.apply( "hallo" ) ).onProperty( "id" ).containsOnly( entity.getId() );
+		assertThat( search.apply( "au revoir" ) ).onProperty( "id" ).containsOnly( entity2.getId() );
+
+		dao.delete( entity );
+		assertThat( search.apply( "bonjour" ) ).onProperty( "id" ).isEmpty();
+		assertThat( search.apply( "hello" ) ).onProperty( "id" ).isEmpty();
+		assertThat( search.apply( "hallo" ) ).onProperty( "id" ).isEmpty();
+		assertThat( search.apply( "au revoir" ) ).onProperty( "id" ).containsOnly( entity2.getId() );
+	}
+
+	@Test
+	public void nonSpringFieldBridge() {
+		Function<String, List<EntityWithSpringAwareBridges>> search = dao::searchNonSpringBridge;
+
+		assertThat( search.apply( NonSpringBridge.PREFIX + InternationalizedValue.HELLO.name() ) ).onProperty( "id" ).isEmpty();
+
+		EntityWithSpringAwareBridges entity = new EntityWithSpringAwareBridges();
+		entity.setInternationalizedValue( InternationalizedValue.HELLO );
+		dao.create( entity );
+		assertThat( search.apply( NonSpringBridge.PREFIX + InternationalizedValue.HELLO.name() ) ).onProperty( "id" ).containsOnly( entity.getId() );
+	}
+
+	@Test
+	public void explicitNonInjectedFieldBridge() {
```

Comment:
@Sanne I think there's something missing here... I'll dive back into it soon and will try to add a proper test if necessary.